### PR TITLE
[backend/frontend] Fix reply to function in animation>mail section

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/979

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -13,9 +13,10 @@
 public class GlobalExceptionHandler {
 
     @ExceptionHandler(ElementNotFoundException.class)
-    public ResponseEntity<ErrorMessage> handleElementNotFoundException() {
-        ErrorMessage message = new ErrorMessage("Not found");
-        return new ResponseEntity<ErrorMessage>(message, HttpStatus.NOT_FOUND);
+    public ResponseEntity<ErrorMessage> handleElementNotFoundException(ElementNotFoundException ex) {
+        ErrorMessage message = new ErrorMessage("Element not found: " + ex.getMessage());
```
Comment (guillaumejparis) [inline openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java:17]: More meaningful logs <3

File: openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -13,9 +13,10 @@
 public class GlobalExceptionHandler {
 
     @ExceptionHandler(ElementNotFoundException.class)
-    public ResponseEntity<ErrorMessage> handleElementNotFoundException() {
-        ErrorMessage message = new ErrorMessage("Not found");
-        return new ResponseEntity<ErrorMessage>(message, HttpStatus.NOT_FOUND);
+    public ResponseEntity<ErrorMessage> handleElementNotFoundException(ElementNotFoundException ex) {
```
Comment (Christian-DONGMO) [inline openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java:16]: What if your exception is null?

File: openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java
Comment (Christian-DONGMO) [inline openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java:1]: 1 - You should use "private final" on all the attributes ofthis class in order to take benefit of lombok annotation (@RequiredArgsConstructor). That will clean some setters
2 - The "mapper" attribute is inherited from RestBehavior, no need to set it again.

**Review by Christian-DONGMO — CHANGES_REQUESTED** [review]

**Comment by Christian-DONGMO** [comment]
For this point : "The response should be logged in OBAS."
I was not able to check this part. I will appreciate if you could show me.

**Comment by guillaumejparis** [comment]
I've added a small ui change for the reply button

**Review by guillaumejparis — APPROVED** [review]

File: openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java (hunk 2 above)
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/config/GlobalExceptionHandler.java:16]: Is not possible, this is the purpose of this kind of handler.
https://www.baeldung.com/exception-handling-for-rest-with-spring

File: openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java:1]: I take action for the second point, the first one wil be in an other step

**Review by Christian-DONGMO — APPROVED** [review]

**Pull request merged**